src/System.Reflection/tests/MethodInfo/MethodInfo_InvokeTests.cs

```diff
@@ -4,252 +4,386 @@
+        public static void TestInvokeMethod_DefaultParameter_NullableIntWithMissingValue()
+        {
+            Assert.Equal(
+                null,
+                (int?)getMethod(typeof(DefaultParameters), "NullableInt").Invoke(new DefaultParameters(), new object[] { Type.Missing }));
+        }
 
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_NullableIntWithExplicitValue()
+        {
+            Assert.Equal(
+                (int?)42,
+                (int?)getMethod(typeof(DefaultParameters), "NullableInt").Invoke(new DefaultParameters(), new object[] { (int?)42 }));
+        }
 
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_EnumParameterWithMissingValue()
+        {
+            Assert.Equal(
+                DefaultParameters.CustomEnum.Default,
+                (DefaultParameters.CustomEnum)getMethod(typeof(DefaultParameters), "Enum").Invoke(new DefaultParameters(), new object[] { Type.Missing }));
+        }
 
-        // Gets MethodInfo object from current class
-        public static MethodInfo getMethod(string method)
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_InterfaceMethodWithMissingValues()
         {
-            return getMethod(typeof(Test), method);
+            Assert.Equal(
+                "1, test, 3.14",
+                (string)getMethod(typeof(IDefaultParameters), "InterfaceMethod").Invoke(
+                    new DefaultParameters(), 
+                    new object[] { Type.Missing, Type.Missing, Type.Missing }));
         }
 
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_InterfaceMethodWithMissingExplicitValues()
+        {
+            Assert.Equal(
+                "101, value, 103.14",
+                (string)getMethod(typeof(IDefaultParameters), "InterfaceMethod").Invoke(
+                    new DefaultParameters(),
+                    new object[] { 101, "value", 103.14m }));
+        }
 
-        //Gets MethodInfo object from a Type
-        public static MethodInfo getMethod(Type t, string method)
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_StaticMethodWithMissingValues()
+        {
+            Assert.Equal(
+                "1, test, 3.14",
+                (string)getMethod(typeof(DefaultParameters), "StaticMethod").Invoke(
+                    null,
+                    new object[] { Type.Missing, Type.Missing, Type.Missing }));
+        }
+
+        [Fact]
+        public static void TestInvokeMethod_DefaultParameter_StaticMethodWithMissingExplicitValues()
+        {
+            Assert.Equal(
+                "101, value, 103.14",
+                (string)getMethod(typeof(DefaultParameters), "StaticMethod").Invoke(
+                    null,
+                    new object[] { 101, "value", 103.14m }));
+        }
+
+        // Gets MethodInfo object from a Type
+        private static MethodInfo getMethod(Type t, string method)
```

Comment:
Unfortunately this is not yet available in .NET Native - we're sharing this test file with .NET Native branches (and I would like to keep it that way).